hi-ml/src/health_ml/run_ml.py

```diff
@@ -154,8 +155,27 @@ def run(self) -> None:
             # Comparison with stored results for cross-validation runs only operates on child run 0. This run
             # has usually already downloaded the results for the other runs, and uploaded files to the parent
             # run context.
+            regression_metrics_str = self.container.regression_metrics
+            regression_metrics = regression_metrics_str.split(',') if regression_metrics_str else None
+
             logging.info("Comparing the current results against stored results")
             if self.is_crossval_disabled_or_child_0():
+                if is_running_in_azure_ml:
+                    if PARENT_RUN_CONTEXT is not None:
+                        df = aggregate_hyperdrive_metrics(
+                            "run",
+                            run=PARENT_RUN_CONTEXT,
+                            keep_metrics=regression_metrics)
+                    else:
+                        df = get_metrics_for_childless_run(
+                            run=RUN_CONTEXT,
+                            keep_metrics=regression_metrics)
+
+                    if len(df) > 0:
```

Comment:
You can use `empty`: https://pandas.pydata.org/pandas-docs/version/0.18/generated/pandas.DataFrame.empty.html